packages/app/src/vis-packs/core/hooks.ts

```diff
@@ -54,17 +54,28 @@ export function useDatasetValue<D extends Dataset<ArrayShape | ScalarShape>>(
 export function useDatasetValues<D extends Dataset<ArrayShape | ScalarShape>>(
   datasets: D[],
   selection?: string
-): Record<string, Value<D>> {
+): Value<D>[];
+
+export function useDatasetValues<D extends Dataset<ArrayShape | ScalarShape>>(
+  datasets: (D | undefined)[],
+  selection?: string
+): (Value<D> | undefined)[];
+
+export function useDatasetValues<D extends Dataset<ArrayShape | ScalarShape>>(
+  datasets: (D | undefined)[],
+  selection?: string
+): (Value<D> | undefined)[] {
```

Comment:
These override allow simplifying the code in `NxValuesFetcher` and getting rid of a couple of custom hooks. We no longer return the datasets' values mapped against their names; instead, we allow `undefined` values arrays in the returned array (in the same place as `undefined` datasets in the input array). The overrides ensure that if we pass an array without `undefined` values, we get a return type without `undefined` values as well.